src/clang-cpp-frontend/clang_cpp_convert.cpp

```diff
@@ -947,29 +947,34 @@ bool clang_cpp_convertert::get_function_body(
 
         if (!init->isBaseInitializer())
         {
-          exprt lhs;
           if (init->isMemberInitializer())
           {
+            exprt lhs;
             // parsing non-static member initializer
             if (get_decl_ref(*init->getMember(), lhs))
               return true;
+
+            build_member_from_component(
+              fd, lhs.id() == "dereference" ? lhs.op0() : lhs);
+
+            exprt rhs;
+            rhs.set("#member_init", 1);
+            if (get_expr(*init->getInit(), rhs))
+              return true;
+
+            initializer = side_effect_exprt("assign", lhs.type());
+            initializer.copy_to_operands(lhs, rhs);
+          }
+          else if (init->isDelegatingInitializer())
+          {
+            if (get_expr(*init->getInit(), initializer))
+              return true;
```

Comment:
IMO, it **might** be missing the assign `exprt`.

like this:
```
initializer = side_effect_exprt("assign", lhs.type());
initializer.copy_to_operands(lhs, rhs);
```

Reply:
And what would be the `lhs`?
Using `this` would feel a bit weird.